servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -8043,6 +8095,13 @@ void normalizeGroupMembers(Group group) {
                 addNormalizedGroupMember(normalizedMembers, member);
             }
         }
+
+        // go through each member and set the principal type
+
+        for (GroupMember member : normalizedMembers.values()) {
```

Comment:
instead of looping over again, can we set the principal type which adding them to "normalizedGroupMembers" above?

Reply:
sure, we'll just need to add some logic to skip duplicates and don't waste lookup operations.